include/Animal.h

```diff
@@ -0,0 +1,33 @@
+#ifndef ANIMAL_H
+#define ANIMAL_H
+
+#include <iostream>
+#include "Location.h"
+
+class Animal
+{
+public:
+	Animal(const std::string& name,const Location& location);
+	~Animal();
+	void printDetails()const;
+	virtual char getInitial() const = 0;
+	std::string getName() const;
+	Location getLocation() const;
+	virtual void step() = 0;
+	void stop();
+	virtual void move();
+	void turnVertically();
```

Comment:
all function should be virtual 
check this
https://stackoverflow.com/questions/2391679/why-do-we-need-virtual-functions-in-c

Reply:
But I don't override them I'm only using the Animal implementation